From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Thu, 14 Aug 2025 02:59:34 +0200
Subject: document-portal/permission-db: Use permissions ID as argument
 instead of app

We only change this in the public API, not to clutter the diff with
variables renames, but the permission DB should now handle just apps
permissions IDs

Origin: https://github.com/flatpak/xdg-desktop-portal/pull/1764
---
 document-portal/permission-db.c | 4 ++--
 document-portal/permission-db.h | 8 ++++----
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/document-portal/permission-db.c b/document-portal/permission-db.c
index 7c0284e..d3f35c2 100644
--- a/document-portal/permission-db.c
+++ b/document-portal/permission-db.c
@@ -49,11 +49,11 @@ struct PermissionDb
 
   gboolean   dirty;
 
-  /* Map id => GVariant (data, sorted-dict[appid->perms]) */
+  /* Map id => GVariant (data, sorted-dict[app_permissions_id->perms]) */
   GvdbTable  *main_table;
   GHashTable *main_updates;
 
-  /* (reverse) Map app id => [ id ]*/
+  /* (reverse) Map app permissions id => [ id ]*/
   GvdbTable  *app_table;
   GHashTable *app_additions;
   GHashTable *app_removals;
diff --git a/document-portal/permission-db.h b/document-portal/permission-db.h
index b2965ba..52ca339 100644
--- a/document-portal/permission-db.h
+++ b/document-portal/permission-db.h
@@ -45,7 +45,7 @@ PermissionDb *     permission_db_new (const char *path,
 char **        permission_db_list_ids (PermissionDb *self);
 char **        permission_db_list_apps (PermissionDb *self);
 char **        permission_db_list_ids_by_app (PermissionDb  *self,
-                                              const char *app);
+                                              const char *app_permissions_id);
 char **        permission_db_list_ids_by_value (PermissionDb *self,
                                                 GVariant  *data);
 PermissionDbEntry *permission_db_lookup (PermissionDb  *self,
@@ -79,7 +79,7 @@ void            permission_db_entry_unref (PermissionDbEntry *entry);
 GVariant *      permission_db_entry_get_data (PermissionDbEntry *entry);
 const char **   permission_db_entry_list_apps (PermissionDbEntry *entry);
 const char **   permission_db_entry_list_permissions (PermissionDbEntry *entry,
-                                                      const char     *app);
+                                                      const char     *app_permissions_id);
 GString *       permission_db_entry_print_string (PermissionDbEntry *entry,
                                                   GString        *string);
 
@@ -87,10 +87,10 @@ PermissionDbEntry  *permission_db_entry_new (GVariant *data);
 PermissionDbEntry  *permission_db_entry_modify_data (PermissionDbEntry *entry,
                                                      GVariant       *data);
 PermissionDbEntry  *permission_db_entry_set_app_permissions (PermissionDbEntry *entry,
-                                                             const char     *app,
+                                                             const char     *app_permissions_id,
                                                              const char    **permissions);
 PermissionDbEntry  *permission_db_entry_remove_app_permissions (PermissionDbEntry *entry,
-                                                                const char        *app);
+                                                                const char        *app_permissions_id);
 
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (PermissionDb, g_object_unref)
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (PermissionDbEntry, permission_db_entry_unref)
